packages/api/cms-api/src/page-tree/blocks/dam-file-download-link.block.ts

```diff
@@ -0,0 +1,108 @@
+import {
+    AnnotationBlockMeta,
+    BlockContext,
+    BlockData,
+    BlockField,
+    BlockInput,
+    BlockMetaField,
+    BlockMetaFieldKind,
+    createBlock,
+    inputToData,
+    TraversableTransformResponse,
+} from "@comet/blocks-api";
+import { IsEnum, IsOptional, IsUUID } from "class-validator";
+import { FilesService } from "src/dam/files/files.service";
+
+export enum OpenFileTypeMethod {
+    NEW_TAB = "NEW_TAB",
+    DOWNLOAD = "DOWNLOAD",
+}
+
+class DamFileDownloadLinkBlockData extends BlockData {
```

Comment:
Yes, since the block doesn't build upon `PixelImageBlock` or the `SvgImageBlock`. 

@F10Ragger Please add `indexData()` similarly to 

https://github.com/vivid-planet/comet/blob/8ed9698122b8bf9eb011672fd6253d7fc1fae71b/packages/api/cms-api/src/blocks/PixelImageBlock.ts#L84-L97

`indexData()` is used to build our block index so we can reconstruct which files are used by which document